test-framework/src/main/java/org/keycloak/test/framework/server/KeycloakTestServer.java

```diff
@@ -1,13 +1,69 @@
 package org.keycloak.test.framework.server;
 
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.keycloak.it.TestProvider;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 public interface KeycloakTestServer {
 
-    void start(List<String> rawOptions);
+    void start(List<String> rawOptions, Set<Class<? extends TestProvider>> providerModules);
 
     void stop();
 
     String getBaseUrl();
 
+    default <T> T createProviderJar(TestProvider provider, Class<T> returnType) {
```

Comment:
Extract this into a utility class or something, this class is getting way to big